scripts/galah/build_parent_sample.py

```diff
@@ -1,391 +1,589 @@
-        results[i]['spectrum_flux'] = np.pad(results[i]['spectrum_flux'], (0, max_length - len(results[i]['spectrum_flux'])), mode='constant')
-        results[i]['spectrum_flux_ivar'] = np.pad(results[i]['spectrum_flux_ivar'], (0, max_length - len(results[i]['spectrum_flux_ivar'])), mode='constant')
-        results[i]['spectrum_lsf'] = np.pad(results[i]['spectrum_lsf'], (0, max_length - len(results[i]['spectrum_lsf'])), mode='constant')
-        results[i]['spectrum_lsf_sigma'] = np.pad(results[i]['spectrum_lsf_sigma'], (0, max_length - len(results[i]['spectrum_lsf_sigma'])), mode='constant')
-        results[i]['spectrum_lambda'] = np.pad(results[i]['spectrum_lambda'], (0 , max_length - len(results[i]['spectrum_lambda'])), mode='constant', constant_values=-1)
-        results[i]['spectrum_norm_flux'] = np.pad(results[i]['spectrum_norm_flux'], (0, max_length - len(results[i]['spectrum_norm_flux'])), mode='constant')
-        results[i]['spectrum_norm_ivar'] = np.pad(results[i]['spectrum_norm_ivar'], (0, max_length - len(results[i]['spectrum_norm_ivar'])), mode='constant')
-        results[i]['spectrum_norm_lambda'] = np.pad(results[i]['spectrum_norm_lambda'], (0, max_length - len(results[i]['spectrum_norm_lambda'])), mode='constant', constant_values=-1)
-
-    # Aggregate all spectra into an astropy table
-    spectra = Table({k: np.vstack([d[k] for d in results])
-                     for k in results[0].keys()})
-    
-    for feature in _FLOAT_FEATURES:
-        spectra[feature] = spectra[feature].flatten()
-    
-    catalog = catalog[['object_id'] + _PARAMETERS]
-
-    catalog = join(catalog, spectra, keys='object_id', join_type='right')
-
-    # Making sure we didn't lose anyone
-    assert len(catalog) == len(spectra), "There was an error in the join operation, probably some spectra files are missing"
-
-    # Save all columns to disk in HDF5 format
-    with h5py.File(output_filename, 'w') as hdf5_file:
-        for key in catalog.colnames:
-            hdf5_file.create_dataset(key, data=catalog[key])
-    return 1
+    max_length = max([len(s["spectrum_flux"]) for s in spectra])
+    num_processed = len(spectra)
+
+    for i in range(len(spectra)):
+        for k in [
+            "spectrum_flux",
+            "spectrum_ivar",
+            "spectrum_lsf",
+            "spectrum_lsf_sigma",
+            "spectrum_norm_flux",
+            "spectrum_norm_ivar",
+        ]:
+            spectra[i][k] = np.pad(
+                spectra[i][k], (0, max_length - len(spectra[i][k])), mode="constant"
+            )
+        for k in ["spectrum_lambda", "spectrum_norm_lambda"]:
+            spectra[i][k] = np.pad(
+                spectra[i][k],
+                (0, max_length - len(spectra[i][k])),
+                mode="constant",
+                constant_values=-1,
+            )
+
+    spectra = collate(spectra)
+
+    os.makedirs(output_dir, exist_ok=True)
+    with h5py.File(os.path.join(output_dir, "001-of-001.hdf5"), "a") as f:
```

Comment:
I think we should be storing these as chunks in each hdf5 file. I.e. limited to N samples per HDF5 file and actually use the indicies? Even if for some healpix folders this chagnes nothing, it likely will reduce errors later.

Reply:
Yes we have this mechanism if we ever run into files that start to be way too big. But for now the cross matching logic assumes there is a single hdf5 file per heal pix region. So for Galah I think we can just have single files, they are probably not that big

Reply:
Okay, I have implemented the splitting/chunking with a default of a one chunk per healpix. So it should work with the current cross matching logic but we can also update this in the future.

Reply:
Also yeah for GALAH it doesn't seem to matter too much, for a version I've already built the healpix with the largest number of objects has 2335 rows.